src/System.Security.Cryptography.Cng/src/System/Security/Cryptography/CngKey.Create.cs

```diff
@@ -48,7 +49,16 @@ public static CngKey Create(CngAlgorithm algorithm, string keyName, CngKeyCreati
 
             errorCode = Interop.NCrypt.NCryptFinalizeKey(keyHandle, 0);
             if (errorCode != ErrorCode.ERROR_SUCCESS)
-                throw errorCode.ToCryptographicException();
+            {
+                Exception e = errorCode.ToCryptographicException();
+                if (CngKey.IsECNamedCurve(algorithm.Algorithm) && 
+                    (errorCode == ErrorCode.NTE_INVALID_PARAMETER || errorCode == ErrorCode.NTE_NOT_SUPPORTED))
+                {
+                    // Curve is not supported
+                    throw new PlatformNotSupportedException(SR.Cryptography_InvalidCurveOid, e);
```

Comment:
Shouldn't this be "InvalidCurve" (or "CurveNotSupported" post a rename)?  The error isn't that FriendlyName was null/empty, which is what InvalidCurveOid reads as.